src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -99,6 +102,19 @@ def __init__(
         self.upgrade_strategy = upgrade_strategy
         self._result = None  # type: Optional[Result]
 
+    def _check_wheel_is_supported(self, link):
+        # type: (Optional[Link]) -> None
+        if not (link and link.is_wheel):
+            return
+        wheel = Wheel(link.filename)
+        tags = compatibility_tags.get_supported()
```

Comment:
This doesn't work because it doesn't respect `--python` (see `test_download__python_version` for an example)